cupyx/scipy/sparse/coo.py

```diff
@@ -116,8 +116,17 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
 
             self.has_canonical_format = False
 
+        elif base.isdense(arg1):
+            # Note: This implementation is not efficeint, as it first
+            # constructs a sparse matrix with csr fromat, then converts it to
+            # coo format.
```

Comment:
How about using `nonzero`?
```python
dense = cupy.atleast_2d(arg1)
row, col = dense.nonzero()
data = dense[row, col]
```
This is not very efficient anyway, but it looks memory-efficient and might be faster.

Reply:
Thanks! Surely using nonzero() would be more efficient. I'll use that.